crates/bevy_reflect/src/serde/deserialize_reflect.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{FromType, Reflect, TypeRegistry};
+use serde::{Deserialize, Deserializer};
+
+/// Trait used to provide finer control when deserializing a reflected type with one of
+/// the reflection deserializers.
+///
+/// This trait is the reflection equivalent of `serde`'s [`Deserialize`] trait.
+/// The main difference is that this trait provides access to the [`TypeRegistry`],
+/// which means that we can use the registry and all its stored type information
+/// to deserialize our type.
+///
+/// This can be useful when writing a custom reflection deserializer where we may
+/// want to handle parts of the deserialization process, but temporarily pass control
+/// to the standard reflection deserializer for other parts.
+///
+/// Without this trait and its associated [type data], such a deserializer would have to
+/// write out all of the deserialization logic itself, possibly including
+/// unnecessary code duplication and trivial implementations.
+///
+/// # Implementors
+///
+/// In order for this to work with the reflection deserializers like [`TypedReflectDeserializer`]
+/// and [`UntypedReflectDeserializer`], implementors should be sure to register the
+/// [`ReflectDeserializeReflect`] type data.
+/// This can be done [via the registry] or by adding `#[reflect(DeserializeReflect)]` to
+/// the type definition.
+///
+/// Note that this trait has a blanket implementation for all types that implement
+/// [`Reflect`] and [`Deserialize`].
```

Comment:
That blanket implementation just falls back on the regular `serde::Deserialize`, right? (and ignores the type registry) Maybe add a note about that?

Reply:
Will do!